### PR TITLE
fixed toggle for social field

### PR DESCRIPTION
**Pull request by elimelt**
Author: Elijah

## What changes were made?

the toggle logic was backwards for isPrivate

### CODE REVIEW

File: src/components/MemberProfileEdit.tsx [hunk 1]
```diff
@@ -110,10 +110,12 @@ const MemberProfileEdit: React.FC<MemberProfileEditProps> = ({
       ...prev,
       [field]: {
         ...(prev[field] as SocialField),
-        isPrivate: !(prev[field] as SocialField)?.isPrivate,
+        isPrivate: !!(prev[field] as SocialField)?.isPrivate === false,
```
Comment (ShawnCollinge) [inline src/components/MemberProfileEdit.tsx:113]: wut

File: src/components/MemberProfileEdit.tsx [hunk 2]
```diff
@@ -110,10 +110,12 @@ const MemberProfileEdit: React.FC<MemberProfileEditProps> = ({
...
       },
     }));
   };
+
+  console.log(profile);
```
Comment (ShawnCollinge) [inline src/components/MemberProfileEdit.tsx:118]: are you trolling?

Comment (elimelt) [reply src/components/MemberProfileEdit.tsx]: yes

File: src/components/MemberProfileEdit.tsx (hunk 1 above)
Comment (elimelt) [reply src/components/MemberProfileEdit.tsx]: fixed

File: src/components/MemberProfileEdit.tsx [hunk 3]
```diff
@@ -102,18 +102,26 @@ const MemberProfileEdit: React.FC<MemberProfileEditProps> = ({
...
+    if (!isSocialFieldKey(baseField)) return;
 
     setProfile((prev) => ({
       ...prev,
-      [field]: {
-        ...(prev[field] as SocialField),
-        isPrivate: !(prev[field] as SocialField)?.isPrivate,
+      [baseField]: {
+        ...prev[baseField],
+        isPrivate: !Boolean(prev[baseField]?.isPrivate),
```
Comment (ShawnCollinge) [inline src/components/MemberProfileEdit.tsx:120]: so isPrivate is set to the opposite of isPrivate? I'm confused lol.

**Review by ShawnCollinge — APPROVED** [review]

**Pull request merged**